### DISCUSSION TITLE
addVariant custom class background color not working

### DISCUSSION THREAD

**Discussion by arashsheyda** [tailwindlabs/tailwindcss#10869] [Help] [answered]
I want it to work like `dark:`

in my root html i have a class named `coffee` and i want to have classes like `coffee:bg-red-200`

I added the below code to my `tailwind.config` file but it didn't work
```
addVariant('coffee', [
    '@media (prefers-color-scheme: coffee)',
])
 ```

**Comment by wongjn** [comment, marked as answer]
The only valid values for [`prefers-color-scheme` is `light` or `dark`](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme) (as these are analogous to the theme settings found on many OSes), so `coffee` would not work here. The closest you can get with `dark:` is to use the the `class` strategy, along with a theme switcher built with JavaScript.

**Reply by wongjn** [reply]
> so i have to add the variants manully right? how can i do that? what way would you suggest? thank you[^1]

You could use a class on some high-level element, like the `<body>` or `<html>`, for example:
```js
addVariant('coffee', '.coffee :is(&)');
// or
addVariant('coffee', '&:is(.coffee *)');
// etc.
```
```html
<html class="coffee">
  <!-- … -->
  <body class="coffee:bg-red-200">
```

You'd use a theme switcher built with JavaScript to toggle the class on the high-level element.

[^1]: https://github.com/tailwindlabs/tailwindcss/discussions/10869#discussioncomment-5434162

**Comment by arashsheyda** [comment]
@wongjn 
so i have to add the variants manully right? how can i do that? what way would you suggest? thank you

**Reply by wongjn** [reply]
Replied at https://github.com/tailwindlabs/tailwindcss/discussions/10869#discussioncomment-5434216 to keep the reply chain together.
